apps/api/src/services/data-gathering/data-gathering.processor.ts

```diff
@@ -102,14 +102,7 @@ export class DataGatheringProcessor {
         }
 
         // Count month one up for iteration
-        currentDate = new Date(
-          Date.UTC(
-            getYear(currentDate),
-            getMonth(currentDate),
-            getDate(currentDate) + 1,
-            0
-          )
-        );
+        currentDate.setDate(currentDate.getDate() + 1);
```

Comment:
To be honest, I'm not sure if this swap would work, I'm trying to figure it out.
When using the currentDate.setDate + currentDate.getDate, there's no changes at the Time Zone, it's a neutral sum of one day, but when I've tried using the addDays function, it converted currentDate for the next day midnight at local time, as in the picure below.

The thing is: I do think we can neglect this timezone difference in the loop condition, because usually it in the worst case it won't retrieve the current day from API, but it'll when the loop runs in the following day. Another workarround would just set the limitDate as today + 1 day, so we'd be sure that it'll get every single day available at API, I'd also add a iteration limit, like 100 years of data (100*365), what do you think?!

![Screenshot_20230916_114237](https://github.com/ghostfolio/ghostfolio/assets/3695510/13e38665-e6b4-4fd2-9ce4-524dd71f38e8)